### ISSUE TITLE
Support IPv6 in govc vm.customize

### ISSUE THREAD

**Issue by ttuellmann** [vmware/govmomi#1971]
It would be great to be able to add IPv6 addresses to customization specifications. 

Unfortunately, I do not speak golang (yet). My naive approach to implement IPv6 support `customize.go` failed:

```
for i, ip6 := range cmd.ip6 {
	nic := &spec.NicSettingMap[i]
	nic.Adapter.IpV6Spec.Ip[0] = &types.CustomizationFixedIpV6{IpAddress: ip6, SubnetMask: 64}
}
```

**Comment by dougm** [comment]
Thanks for giving it a try, how did it fail? Can you share your code in a GH branch?  I can take a look and offer suggestions.

**Comment by ttuellmann** [comment]
It panics with an index out of range error. The object model for IPv6 addresses differs from v4 quite a bit, because you can configure multiple addresses at once through the API.

```
$ ~/go/bin/govc vm.customize -vm test-vm -ip6 2001:db8::1
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/vmware/govmomi/govc/vm.(*customize).Run(0xc0001c2000, 0x1fb3100, 0xc00003c0a0, 0xc000032900, 0x0, 0x0)
	/Users/tt/go/src/github.com/vmware/govmomi/govc/vm/customize.go:241 +0x14e9
github.com/vmware/govmomi/govc/cli.Run(0xc000032250, 0x5, 0x5, 0xc00010a058)
	/Users/tt/go/src/github.com/vmware/govmomi/govc/cli/command.go:165 +0x5de
main.main()
	/Users/tt/go/src/github.com/vmware/govmomi/govc/main.go:98 +0x64
```

There is a minimal example in https://github.com/ttuellmann/govmomi/commit/ef63d1e556c12c2d0e4d75768ac3737cd3bc1c55

Thanks!

**Comment by dougm** [comment]
Sorry for the delayed response.. switch to using append should avoid that problem:
```go
nic.Adapter.IpV6Spec.Ip = append(nic.Adapter.IpV6Spec.Ip, &types.CustomizationFixedIpV6{IpAddress: ip6, SubnetMask: 64})
```

**Comment by ttuellmann** [comment]
This time I have to apologise for the delay. With a lot of help from @hreese , we have compiling code at https://github.com/hreese/govmomi/commit/3913964376ecdc375a783f499497f40f7a815aa6

**Comment by dougm** [comment]
Fixed in #2151
